crates/core/src/chunking.rs

```diff
@@ -68,20 +68,21 @@ impl ChunkEndpoint {
     #[instrument(level = "trace", skip(self))]
     pub async fn get_unchunkified(&self, inv_id: &str) -> anyhow::Result<Vec<u8>> {
         let mut result = Vec::new();
-        let store = self.create_or_reuse_store().await?;
-        debug!(invocation_id = %inv_id, "chunkify starting to receive");
-        let mut obj = store
-            .get(inv_id)
+        let store = self
+            .create_or_reuse_store()
             .await
-            .map_err(|e| anyhow!(e))
-            .context("error starting to receive chunked stream for inv {inv_id}")?;
-        let r = obj.read_to_end(&mut result);
-        r.map_err(|e| anyhow!(e).context("error receiving chunked stream for inv {inv_id}"))
+            .map_err(|e| anyhow!(e).context("failed to get object store"))?;
+        debug!(invocation_id = %inv_id, "chunkify starting to receive");
+        let mut obj = store.get(inv_id).await.map_err(|e| {
+            anyhow!(e).context("failed to receive chunked stream for inv {inv_id}: {e}")
```

Comment:
These strings will not be formatted, you should use `with_context` here (or `format!` if you're using `anyhow!` macro)

Can't we directly call `context` on the returned error here?

Reply:
Thanks 👍